app/modules/listings/components/listing_element_comment.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts">
+  import autosize from 'autosize'
+  import { userContent } from '#app/lib/handlebars_helpers/user_content'
+  import { icon } from '#app/lib/icons'
+  import { getActionKey } from '#app/lib/key_events'
+  import Spinner from '#components/spinner.svelte'
+  import { updateElement } from '#listings/lib/listings'
+  import { I18n, i18n } from '#user/lib/i18n'
+
+  export let isCreatorMainUser, flash, element
+
+  let isEditMode, validating
+  let newComment = element.comment
+
+  async function validate () {
+    validating = _updateElement()
+    await validating
+    toggleEditMode()
+  }
+
+  function onKeyDown (e) {
+    const key = getActionKey(e)
+    if (key === 'esc') isEditMode = false
+    else if (e.ctrlKey && key === 'enter') validate()
+    e.stopPropagation()
+  }
+
+  function toggleEditMode () {
+    isEditMode = !isEditMode
+    flash = null
+  }
+
+  $: comment = element.comment
+  async function _updateElement () {
+    flash = null
+    try {
+      const updatedElement = await updateElement({ id: element._id, comment: newComment })
+      element = updatedElement
+    } catch (err) {
+      flash = err
+    }
+  }
+</script>
+<div class="listing-element-comment">
+  {#if isEditMode}
+    <label>
+      {I18n('comment')}
+      <textarea
+        type="text"
+        bind:value={newComment}
+        use:autosize
+        on:keydown={onKeyDown}
+      />
+    </label>
+    <div class="button-group-right">
+      <button
+        class="tiny-button cancel"
+        on:click={toggleEditMode}
+      >
+        {i18n('cancel')}
+      </button>
+      <button
+        class="tiny-button validate success"
+        title={I18n('validate')}
+        on:click={validate}
+      >
+        {@html icon('check')}
+        {I18n('validate')}
+        {#await validating}
+          <Spinner />
+        {/await}
+      </button>
```

Comment:
[non-blocking]
```suggestion
      <button
        class="tiny-button validate success"
        title={I18n('validate')}
        disabled={validating != null}
        on:click={validate}
      >
        {#if validating}
          <Spinner />
        {:else}
          {@html icon('check')}
        {/if}
        {I18n('validate')}
      </button>
```

Two things there:
- Show the spinner where the icon is to prevent an unesthetical width jump
- Disable the button while validating to prevent a double request. This requires being able to use `validating` as a boolean, which can be done either by doing `validating = null` once the promise resolved, or by turning `validating` into a boolean. I tend to prefer this boolean approach these days, which is a reversal on an earlier opinion of mine, sorry for that :sweat_smile:

Reply:
> I tend to prefer this boolean approach these days

Any reason behind this reversal? I still think the boolean is more "manipulable"

addressed in 80b3f45ff

Reply:
The main point for preferring booleans IMO is the ability to pass it to the `disabled` attribute: getting a boolean from a promise is surprisingly not straight-forward (I wish `Promise.isResolved` was a thing, but it isn't)